src/workerd/api/worker-rpc.c++

```diff
@@ -2,19 +2,330 @@
+  kj::Promise<void> call(CallContext callContext) override {
+    auto methodName = kj::heapString(callContext.getParams().getMethodName());
+    auto serializedArgs = callContext.getParams().getSerializedArgs().getV8Serialized().asBytes();
+
+    // We want to fulfill the callPromise so customEvent can continue executing
+    // regardless of the outcome of `call()`.
+    KJ_DEFER(callFulfiller->fulfill(););
+
+    // Try to execute the requested method.
+    co_return co_await ctx.run(
+        [this, methodName=kj::mv(methodName), serializedArgs=kj::mv(serializedArgs),
+          entrypointName=entrypointName, callContext] (Worker::Lock& lock) mutable ->
+            kj::Promise<void> {
+
+      jsg::Lock& js = lock;
+      if (!FeatureFlags::get(js).getJsRpc()) {
+        // JS RPC is not enabled on the server side, we cannot call any methods.
+        auto rpcDisabled =
+            "The receiving Worker does not allow its methods to be called over RPC."_kjc;
+        JSG_FAIL_REQUIRE(Error, rpcDisabled);
+      }
+
+      // We will try to get the function, if we can't we'll throw an error to the client.
+      auto fn = tryGetFn(lock, entrypointName, ctx, methodName);
+
+      // We have a function, so let's call it and serialize the result for RPC.
+      // We account for both the JS method throwing and our serialization attempt failing.
+      // If the function returns a promise we will wait for the promise to finish so we can
+      // serialize the result.
+      return ctx.awaitJs(js, callFuncAndSerialize(js, fn, serializedArgs).then(js,
+          ctx.addFunctor([callContext](jsg::Lock& js, SerializedResult result) mutable {
+        KJ_SWITCH_ONEOF(result) {
+          KJ_CASE_ONEOF(ok, Returned) {
+            auto builder = callContext.initResults(capnp::MessageSize {
+                ok.serializedValue.size() / 8 + 8, 0 });
+            builder.initValue().setV8Serialized(kj::mv(ok.serializedValue));
+          }
+          KJ_CASE_ONEOF(err, ThrewException) {
+            auto builder = callContext.initResults(capnp::MessageSize {
+                err.serializedException.size() / 8 + 8, 0 });
+            builder.initException().setV8Serialized(kj::mv(err.serializedException));
+          }
         }
-    ));
+      })));
+    });
+  }
+
+  KJ_DISALLOW_COPY_AND_MOVE(JsRpcTargetImpl);
+
+private:
+  // The following names are registered by the Workers Runtime and cannot be called over RPC.
+  bool isRegisteredName(kj::StringPtr name) {
+    if (name == "fetch"_kjc ||
+        name == "connect"_kjc ||
+        name == "alarm"_kjc ||
+        name == "webSocketMessage"_kjc ||
+        name == "webSocketClose"_kjc ||
+        name == "webSocketError"_kjc) {
+      return true;
+    }
+    return false;
+  }
+
+  // If the `methodName` is a known public method, we'll return it.
+  inline v8::Local<v8::Function> tryGetFn(
+      Worker::Lock& lock,
+      kj::Maybe<kj::StringPtr> entrypointName,
```

Comment:
This parameter seems redundant since it's the same as the class member.

Reply:
Sorry, mind expanding on this? We're using it to get the exported handler, which I'm not sure how else we'd get?

Reply:
`tryGetFn` is a method of `JsRpcTargetImpl`. This class has a member `kj::Maybe<kj::StringPtr> entrypointName;`, and indeed the value of that member is what is being passed as the parameter to `tryGetFn()`. Why not just use the member directly from inside `tryGetFn()`?

Reply:
Oh wow... that went way over my head :neutral_face:. I was thinking about the JS class instead of C++ 

https://github.com/cloudflare/workerd/compare/6b11989af61f7ca72c4fbd5c2bbd89d377970571..004425cb34c59b5488a066bdcdf72332f49fc8a4